File: app/src/main/kotlin/me/tomasan7/jecnadesktop/repository/GradesRepository.java
package me.tomasan7.jecnadesktop.repository;

import me.tomasan7.jecnadesktop.data.Grades;

import java.util.concurrent.CompletableFuture;

/**
 * Retrieves {@link Grades} from any kind of data source.
 */
public interface GradesRepository
{
	Grades queryGrades ();

	default CompletableFuture<Grades> queryGradesAsync ()
	{
		return CompletableFuture.supplyAsync(this::queryGrades);
	}
}

File: src/main/kotlin/me/tomasan7/jecnadesktop/repository/GradesRepository.kt
package me.tomasan7.jecnadesktop.repository

import me.tomasan7.jecnadesktop.data.Grades
import me.tomasan7.jecnadesktop.data.SchoolYear

/**
 * Retrieves [Grades] from any kind of data source.
 */
interface GradesRepository
{
    suspend fun queryGrades(): Grades

    /**
     * @param schoolYear The [SchoolYear] to get the [Grades] for.
     * @param firstHalf The school year half to get the [Grades] for. `true` for first half and `false` for second half.
     */
    suspend fun queryGrades(schoolYear: SchoolYear, firstHalf: Boolean): Grades
}